src/Microsoft.AspNet.Mvc.TagHelpers/InputTagHelper.cs

```diff
@@ -236,25 +236,27 @@ private void GenerateCheckBox(ModelExplorer modelExplorer, TagHelperOutput outpu
                 }
             }
 
-            var tagBuilder = Generator.GenerateCheckBox(
+            var checkBoxTag = Generator.GenerateCheckBox(
                 ViewContext,
                 modelExplorer,
                 For.Name,
                 isChecked: null,
                 htmlAttributes: htmlAttributes);
-            if (tagBuilder != null)
+            if (checkBoxTag != null)
             {
                 // Do not generate current element's attributes or tags. Instead put both <input type="checkbox"/> and
                 // <input type="hidden"/> into the output's Content.
                 output.Attributes.Clear();
                 output.TagName = null;
 
-                output.Content.Append(tagBuilder.ToHtmlContent(TagRenderMode.SelfClosing));
+                checkBoxTag.TagRenderMode = TagRenderMode.SelfClosing;
```

Comment:
I don't think combining a `bool` and `TagBuilderMode` makes things any simpler. I'd like to stick to having one way to set the mode. Easier to document, easier to understand.

But one thing that I think would be nice is if it would be possible to set the mode in the generator as opposed to out here.